### ISSUE TITLE
Does it work with the Go2 pro?

### ISSUE THREAD

**Issue by marco-teixeira** [MAVProxyUser/YushuTechUnitreeGo1#13]
Does it work with the Go2 pro?

**Comment by MAVProxyUser** [comment]
what is "it"? Does what specifically work with Go2 pro? 
You are commenting on a Go1 based repo.

**Comment by marco-teixeira** [comment]
I would like to know if there is something similar for the Go2 pro.

The EDU version is very expensive. From what I could see, this repository allows you to control the Go1 pro in several ways, but I didn't identify anything similar to the Go2 pro.

Apparently the Go1 pro version has been discontinued. I'm just identifying the go 2 pro version for acquisition.

**Comment by MAVProxyUser** [comment]
Got ya! We know some of the architecture is the same, but they have moved away from Jetson's on board, and the architecture has completely changed. They are using ROS2 with DDS now, but most of the same low level encflag logic seems to remain. We need to get access to a do so we can work on the Faux Level compatibility... https://github.com/Bin4ry/free-dog-sdk

**Comment by marco-teixeira** [comment]
thanks for your answer.

I have never worked with Go1 and Go2, but I work with ROS1 and ROS2.

I don't understand when unitree says that only the EDU version is programmable. I can't access the Go(1,2) onboard computer and program them via ROS topics? Do they block or hinder this access in any way in non-EDU versions?

I'm from Brazil, and I have few resources to invest in this equipment, I'm afraid of buying the Go2 pro version and not being able to use it for research.

I found this repository, but I believe it only works in the EDU version:
https://github.com/unitreerobotics/unitree_ros2

**Comment by MAVProxyUser** [comment]
They lied about programability of non EDU dogs to increase their profits. They have I believe on recent firmware imposed artificial limitations.

**Comment by xifen523** [comment]
> thanks for your answer.
> 
> I have never worked with Go1 and Go2, but I work with ROS1 and ROS2.
> 
> I don't understand when unitree says that only the EDU version is programmable. I can't access the Go(1,2) onboard computer and program them via ROS topics? Do they block or hinder this access in any way in non-EDU versions?
> 
> I'm from Brazil, and I have few resources to invest in this equipment, I'm afraid of buying the Go2 pro version and not being able to use it for research.
> 
> I found this repository, but I believe it only works in the EDU version: https://github.com/unitreerobotics/unitree_ros2

Have you found some way to get programming access on the non-EDU version yet?

**Comment by MAVProxyUser** [comment]
Yes... the Go2 has been rooted about 6 months ago. Our slack group has the discussion. 
This invite link is valid for 18 more days. https://join.slack.com/t/robotdogs/shared_invite/zt-2kdvfdwt0-29NnVE08iV8g0PpkYhQmhg

**Comment by MX-LZX** [comment]
I am currently a graduate student researching quadruped robots, and I’m also an enthusiast in this field. My research involves the Unitree Go1 Pro robot, focusing on NMPC (Nonlinear Model Predictive Control) and WBC (Whole-Body Control). Recently, I attempted to use external pressure sensors to collect foot-end data and provide feedback to the main control unit, but I encountered some issues. I would love to join your Slack group to learn together if that's possible.
My email addresses are: 936915881mxlzy@gmail.com or mengxinlzy936@163.com.

**Comment by MAVProxyUser** [comment]
> I am currently a graduate student researching quadruped robots, and I’m also an enthusiast in this field. My research involves the Unitree Go1 Pro robot, focusing on NMPC (Nonlinear Model Predictive Control) and WBC (Whole-Body Control). Recently, I attempted to use external pressure sensors to collect foot-end data and provide feedback to the main control unit, but I encountered some issues. I would love to join your Slack group to learn together if that's possible. My email addresses are: [936915881mxlzy@gmail.com](mailto:936915881mxlzy@gmail.com) or [mengxinlzy936@163.com](mailto:mengxinlzy936@163.com).

Invite sent to your gmail account @MX-LZX

**Comment by jaysplains** [comment]
Hi there can I get an invite too as I'm looking to start programming the go2 pro I can be emailed at jaysplains@icloud.com thank you

**Comment by MAVProxyUser** [comment]
@Jaysplains this link is valid for 26 more days. https://join.slack.com/t/robotdogs/shared_invite/zt-30ol5tib4-HU5ykAdMfLuhIjJY1ZTg0A

**Comment by YoucefAouf** [comment]
Hi, I’m currently working with the Go1 and am planning to start with the Go2 soon. Could you send an invite to ylibzou@gmail.com?

**Comment by yongqingzs** [comment]
Hi, I have successfully deployed the mpc+wbc controller on go1 and now want to try it on go2. Could you send an invite to: c1106381878@gamil.com, thank you.
